cargo-miri/bin.rs

```diff
@@ -888,7 +920,15 @@ fn phase_rustc(mut args: impl Iterator<Item = String>, phase: RustcPhase) {
                 cmd.arg("--emit=metadata");
             }
 
-            cmd.args(&env.args);
+            // Alter the `-o` parameter so that it does not overwrite the JSON file we stored above.
+            let mut args = env.args.clone();
+            for i in 0..args.len() {
+                if args[i] == "-o" {
+                    args[i + 1].push_str("_miri");
```

Comment:
```suggestion
                    args[i + 1].push_str(".miri");
```
Seems more consistent with .stdin below.

Does this lead to loads of stale files being created?

Reply:
Consistency is good.

I can't find any stale files from this. I know that some are definitely emitted into the temp directories that rustdoc creates, but it seems to know to unlink them as well. I feel like they should be emitted in another scenario, but I can't find any under `target/`.